hibernate-core/src/main/java/org/hibernate/boot/model/naming/CamelCaseToUnderscoresNamingStrategy.java

```diff
@@ -86,7 +86,15 @@ protected boolean isCaseInsensitive(JdbcEnvironment jdbcEnvironment) {
 	}
 
 	private boolean isUnderscoreRequired(final char before, final char current, final char after) {
-		return Character.isLowerCase( before ) && Character.isUpperCase( current ) && Character.isLowerCase( after );
+		if (Character.isDigit(current)) {
+			if (Character.isDigit(before)) {
+				return false;
+			}
+			if (Character.isLowerCase(before)) {
+				return true;
+			}
+		}
```

Comment:
A bit naively, seems to me that you could simply delete this whole block. What am I missing?

```suggestion
```

Reply:
Yep, sounds good to me. Thanks!